src/test/java/raisetech/student/management/data/domain/StudentDetailTest.java

```diff
@@ -0,0 +1,395 @@
+package raisetech.student.management.data.domain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static raisetech.student.management.testutil.TestDataFactory.completedStudent;
+import static raisetech.student.management.testutil.TestDataFactory.completedStudentCourse;
+import static raisetech.student.management.testutil.TestDataFactory.completedStudentDetail;
+import static raisetech.student.management.testutil.TestDataFactory.enoughStudentCourseOnRegister;
+
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import raisetech.student.management.data.Student;
+import raisetech.student.management.data.StudentCourse;
+import raisetech.student.management.data.domain.validation.OnUpdate;
+
+class StudentDetailTest {
+
+  private static Validator validator;
+
+  @BeforeAll
+  static void setUpValidator() {
+    validator = Validation.buildDefaultValidatorFactory().getValidator();
+  }
+
+  @Test
+  void 受講生リクエストボディ検証_すべてのフィールドが登録時に期待する値をもつとき入力チェックにかからないこと() {
+    // Arrange:
+    int studentId = 0;
+    int courseId = 0;
+    StudentDetail validStudentDetail = new StudentDetail(
+        completedStudent(studentId), List.of(enoughStudentCourseOnRegister(studentId, courseId))
```

Comment:
makeやcreateなど動詞がある方がよい 👌🏽 
```suggestion
        makeCompletedStudent(studentId), List.of(makeEnoughStudentCourseOnRegister(studentId, courseId))
```